rnd/autogpt_server/autogpt_server/data/graph.py

```diff
@@ -112,47 +128,70 @@ async def get_node(node_id: str) -> Node | None:
 
 
 async def get_graph_ids() -> list[str]:
-    return [graph.id for graph in await AgentGraph.prisma().find_many()]  # type: ignore
+    return [
+        graph.id
+        for graph in await AgentGraph.prisma().find_many(where={"is_template": False})
+    ]  # type: ignore
+
+
+async def get_template_meta() -> list[GraphMeta]:
+    templates = await AgentGraph.prisma().find_many(
+        where={"is_template": True},
+    )
+    if not templates:
+        return []
 
+    return [GraphMeta.from_db(template) for template in templates]  # type: ignore
 
-async def get_graph(graph_id: str) -> Graph | None:
-    graph = await AgentGraph.prisma().find_unique(
-        where={"id": graph_id},
+
+async def get_graph(graph_id: str, is_template: bool = False) -> Graph:
+    graph = await AgentGraph.prisma().find_first_or_raise(
+        where={"id": graph_id, "is_template": is_template},
         include={"AgentNodes": {"include": EXECUTION_NODE_INCLUDE}},  # type: ignore
     )
-    return Graph.from_db(graph) if graph else None
+    return Graph.from_db(graph)
 
 
-async def create_graph(graph: Graph) -> Graph:
+async def create_graph(graph: Graph, is_template: bool = False) -> Graph:
```

Comment:
if we merge is_template into `Graph` this can be intact.

Reply:
Yes, I've done this in the versioning pr